app/assets/javascripts/controllers/FileListCtrl.coffee

```diff
@@ -0,0 +1,42 @@
+# This provides the bFileList directive with some power
+@bruseApp.controller 'FileListCtrl', ['$scope', '$filter', 'FileHandler', ($scope, $filter, FileHandler) ->
+  # since we use 'this' in some function below, we need to make sure they are
+  # use the the controller as the 'this', and not the function
+  self = this;
+  # setup some vars
+  orderBy = $filter('orderBy')
+  $scope.files = []
+  $scope.view_list = true
+
+  # this function gets called from the directive
+  this.init = (element, attrs) ->
+    self.$element = element
+    $scope.name = attrs.name
+    if attrs.files
+      # if file is provided through the directive attributes, use those...
+      $scope.files = attrs.files
+    else
+      # ... otherwise collect them from server
+      FileHandler.collect().then((data) ->
```

Comment:
There is a catch in [`app/assets/javascripts/factories/FileHandler.coffee`](https://github.com/tkomstadius/bruse/blob/feature/file-template/app/assets/javascripts/factories/FileHandler.coffee#L15-L18)